MdePkg/Library/BasePeCoffLib/BasePeCoff.c

```diff
@@ -1054,7 +1054,7 @@ PeCoffLoaderRelocateImage (
     RelocDir = &Hdr.Te->DataDirectory[0];
   }
 
-  if ((RelocDir != NULL) && (RelocDir->Size > 0)) {
+  if ((RelocDir != NULL) && (RelocDir->Size > 0) && (RelocDir->Size - 1 < MAX_UINT32 - RelocDir->VirtualAddress)) {

```

Comment:
Is there a reason extra () were not added to make this change more readable?

Reply:
I can add that but this was the exact patch change on bugzilla - I was trying to not make changes.

Reply:
Ok.  I will approve this version.  Please add the extra () in an additional PR.